build.assets/robotest/nightly_config.sh

```diff
@@ -21,15 +21,21 @@ UPGRADE_MAP[6.2.0]="ubuntu:16"
 
 UPGRADE_VERSIONS=${!UPGRADE_MAP[@]}
 
+# The following breaks a dependency loop. We need upgrade versions to generate tarballs,
+# but we need tarball names to generate the full test config.
+if [[ ${1} == "upgradeexit" ]]  ; then
+    return
+fi
```

Comment:
This still feels like an ugly hack.  How could the code be factored to handle this dependency better?